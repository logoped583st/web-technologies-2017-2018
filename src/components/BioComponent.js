import React from 'react';
import { connect } from "react-redux";


function BioComponent(props) {
    return(
        <div className="Bio">
            <h4>{props.bio}</h4>
        </div>
    );
}

const getState = (state) => {
    return{
        bio: state.reducerUser.bio,
    };
};

export default connect(getState)(BioComponent);