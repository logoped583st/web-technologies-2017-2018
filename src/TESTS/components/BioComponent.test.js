import React from 'react';
import BioComponent from '../../components/BioComponent';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });

describe('BioComponent component test', () => {
    const mockStore = configureStore([thunk]);
    const initialState = {
        reducerUser: 
             "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt" +
            "ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco"+
             "laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in "+
             "voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident,"+
             " sunt in culpa qui officia deserunt mollit anim id est laborum. "
        
    }

    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('BioComponent component rendered succesfully', () => {
        const bioComponent = mount(
            <BioComponent store={store} />
        );
        expect(mountToJson(bioComponent)).toMatchSnapshot();
    });

    it('BioComponent component props match state', () => {
        const bioComponent = shallow(
            <BioComponent store={store} />
        );
        expect(bioComponent.prop('bio')).toBe(initialState.reducerUser.bio);
    })
});