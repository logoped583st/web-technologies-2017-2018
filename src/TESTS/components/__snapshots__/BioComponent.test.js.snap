// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BioComponent component test BioComponent component rendered succesfully 1`] = `
<Connect(BioComponent)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BioComponent
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div
      className="Bio"
    >
      <h4 />
    </div>
  </BioComponent>
</Connect(BioComponent)>
`;
